static/js/components/forms/SiteContentForm.test.tsx

```diff
@@ -1,239 +1,236 @@
-            // @ts-ignore
-            fieldIsVisible.mockImplementation(() => true)
-            // @ts-ignore
-            renameNestedFields.mockImplementation(() => configItem.fields)
-            const values = { file: "courses/file.pdf" }
-            const wrapper = renderInnerForm(editorState, { values })
-            expect(wrapper.find("SiteContentField").exists()).toBe(
-              !isResourceFileField || !isGdriveEnabled
-            )
-          })
-        })
-      })
-
-      it("creates initialValues", () => {
-        const newData = "new data",
-          oldData = "old data"
-        // @ts-ignore
-        newInitialValues.mockReturnValue(newData)
-        // @ts-ignore
-        contentInitialValues.mockReturnValue(oldData)
-        const wrapper = renderForm({ editorState })
-        const initialValues = wrapper.find("Formik").prop("initialValues")
-        if (editorState.adding()) {
-          expect(initialValues).toBe(newData)
-          expect(newInitialValues).toBeCalledWith(configItem.fields, website)
-        } else {
-          expect(initialValues).toBe(oldData)
-          expect(contentInitialValues).toBeCalledWith(
-            content,
-            configItem.fields,
-            website
-          )
-        }
-      })
-
-      //
-      ;[true, false].forEach(isObjectField => {
-        it(`passes an onChange prop for an ${
-          isObjectField ? "object" : "non-object"
-        } field which sets the dirty flag and calls handleChange for formik`, () => {
-          const field = makeWebsiteConfigField({
-            widget: isObjectField ? WidgetVariant.Object : WidgetVariant.String
-          })
-          configItem.fields = [field]
-          // @ts-ignore
-          fieldIsVisible.mockImplementation(() => true)
-          // @ts-ignore
-          renameNestedFields.mockImplementation(() => configItem.fields)
-          const handleChangeStub = sandbox.stub()
-          const form = renderInnerForm(editorState, {
-            handleChange: handleChangeStub
-          })
-          const onChange = form
-            .find(isObjectField ? "ObjectField" : "SiteContentField")
-            .prop("onChange")
-          const event: any = { preventDefault: sandbox.stub() }
-          // @ts-ignore
-          onChange(event)
-          sinon.assert.calledOnceWithExactly(handleChangeStub, event)
-          sinon.assert.calledOnceWithExactly(setDirtyStub, true)
-        })
-      })
+test("it should use an ObjectField when dealing with an Object", () => {
+  const configItem = makeEditableConfigItem()
+  const configField = makeWebsiteConfigField({
+    widget: WidgetVariant.Object
+  })
+  configItem.fields = [configField]
+  const { form } = setup({
+    configItem
+  })
+  const objectField = form.find(ObjectField)
+  expect(objectField.exists()).toBeTruthy()
+  const renamedField = renameNestedFields([configField])[0]
+  expect(objectField.prop("field")).toEqual(renamedField)
+})
+
+test.each`
```

Comment:
I did know know about `test.each`. Cool....!

Reply:
yeah we've been doing it manually, living in ignorance haha

Reply:
the only downside is that you can using a template for the test title, and it's based on a format string instead, but that's not too bad I think